packages/application-shell-connectors/src/components/application-context/normalizers.ts

```diff
@@ -98,23 +98,14 @@ type NormalizedGroupedByResourceType = {
   // E.g. { orders: {...} }
   [key: string]: NormalizedGroupedByPermission | null;
 };
-/**
- * dataFence: {
- *   store: {
- *     orders: {
- *       canManageOrders: { values: ['usa', 'germany'] },
- *       canViewOrders: { values: ['canada'] },
- *     }
- *   }
- * }
- */
 type NormalizedStoresGroupByResourceType = {
+  // E.g [ { group: 'orders', value: 'us', type: 'store', name: 'canManageOrders' } ]
   [key: string]: TStoreDataFence[];
 };
 
 const normalizeAppliedDataFencesForStoresByResourceType = (
   dataFences: Maybe<TStoreDataFence>[]
-) => {
+): NormalizedGroupedByResourceType => {
```

Comment:
The return type can not be inferred anymore? I usually felt it's a good pattern to type inputs and have the return type be inferred.

Reply:
aha.. I like explicit 😅 
but sure, let's get rid of this then

Reply:
I actually did it often at first too cause I felt explicit is good. Then when thinking about flow of types and data through the program one realises that it is redundant. If a wants A and returns AA while b wants AA while returning BB. Then it does matter if you type the return of A to AA or if you have TS infer it. If you change what A returns (implicitly or the explicit) TS will warn you that B‘s input of AA is not fulfilled anymore.